library/network/test/network_interfaces_test.rb

```diff
@@ -306,4 +316,40 @@ def mock_ppp
       expect(subject.Devices[device_type]).to include(device => device_map, "eth1" => {})
     end
   end
+
+  describe "#Write" do
+    let(:data_dir) { File.join(File.dirname(__FILE__), "data") }
+    let(:network_path) { File.join(data_dir, "etc/sysconfig/network") }
+    let(:ifcfg_copy) { File.join(network_path, "ifcfg-copy") }
+    let(:ifcfg_file) { File.join(network_path, "ifcfg-eth1") }
+
+    before do
+      Yast::NetworkInterfaces.CleanCacheRead()
+    end
+
+    around do |example|
+      ::FileUtils.cp(ifcfg_file, ifcfg_copy)
+      change_scr_root(data_dir, &example)
+      ::FileUtils.rm(ifcfg_copy)
+    end
+
+    it "writes interfaces configuration changes to ifcfg files" do
+      devmap = Yast::NetworkInterfaces.devmap("eth1")
+      devmap["DHCLIENT_SET_HOSTNAME"] = "yes"
+      Yast::NetworkInterfaces.Write("")
+      devmap = Yast::NetworkInterfaces.devmap("eth1")
+      expect(devmap["DHCLIENT_SET_HOSTNAME"]).to eq("yes")
+      devmap["DHCLIENT_SET_HOSTNAME"] = nil
+      Yast::NetworkInterfaces.Write("")
+      expect(::FileUtils.compare_file(ifcfg_copy, ifcfg_file)).to eq(true)
+    end
+
+    it "deletes removed interfaces" do
+      size = Yast::NetworkInterfaces.List("").size
+      Yast::NetworkInterfaces.Delete("copy")
+      Yast::NetworkInterfaces.Commit()
+      Yast::NetworkInterfaces.Write("")
+      expect(Yast::NetworkInterfaces.List("").size).to eq(size - 1)
+    end
```

Comment:
test for that reread?

Reply:
Ok, will add an explicit test, I checked in previous test that it was write and modifications were read but it is not mention as documentation